### PR TITLE
guarding jsPDF to only import / run when window is present (aka not d…

### PR DESCRIPTION
**Pull request by floAr**
…uring ssr build)

This guards the jsPDF using in a check if we have a present window (aka running in a browser) this will allow to build server side (ignoring the pdf export here) and then rehydrate it back in when the user runs the webpage

## Related Issue

#2112

## Description

Quick fix until jsPDF fixes theeir own issues (https://github.com/MrRio/jsPDF/issues/2753)

### CODE REVIEW

**Comment by floAr** [comment]
This should fix #2112

File: src/components/m-table-toolbar.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+import Tooltip from "@material-ui/core/Tooltip";
+import Typography from "@material-ui/core/Typography";
+import withStyles from "@material-ui/core/styles/withStyles";
+import { lighten } from "@material-ui/core/styles/colorManipulator";
+import classNames from "classnames";
+import { CsvBuilder } from "filefy";
+import PropTypes, { oneOf } from "prop-types";
+import "jspdf-autotable";
+import * as React from "react";
+const jsPDF = typeof window !== "undefined" ? require("jspdf") : null;
```
Comment (ames89) [inline src/components/m-table-toolbar.js:19]: pretty good

**Comment by mbrn** [comment]
Please resolve conflicts

**Comment by floAr** [comment]
@mbrn & @ames89: All done

File: package.json [hunk 2]
```diff
@@ -82,8 +85,6 @@
     "debounce": "^1.2.0",
     "fast-deep-equal": "2.0.1",
     "filefy": "0.1.10",
-    "jspdf": "1.5.3",
-    "jspdf-autotable": "3.5.3",
```
Comment (gamb) [inline package.json:86]: Do we need to remove these? I just ran into an issue with yarn (berry):
> Module not found: material-table tried to access jsPDF, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Comment (gamb) [reply package.json:86]: Ah my yarn issue turned out to be due to the dynamic `require()` stuff, so re-adding these dependencies wasn't sufficient anyway.

Comment (floAr) [reply package.json]: I think you are right, while merging something went amiss. The idea is to have those depencies isntalled, but only require them when running withing a browser.

Comment (floAr) [reply package.json]: resolved in
f6b13cb

Comment (gamb) [reply package.json]: Nice one @floAr :)

File: package.json [hunk 3]
```diff
@@ -8,6 +8,9 @@
     "dist",
     "types"
   ],
+  "publishConfig": {
```
Comment (mbrn) [inline package.json:11]: Why did you add this?

File: src/components/m-table-toolbar.js [hunk 4]
```diff
@@ -56,15 +58,10 @@ export class MTableToolbar extends React.Component {
   defaultExportCsv = () => {
     const [columns, data] = this.getTableData();
 
-    let fileName = this.props.title || "data";
```
Comment (mbrn) [inline src/components/m-table-toolbar.js:59]: Please rebase your branch. It seems that you removed some new developments.

File: package.json (hunk 3 above)
Comment (floAr) [reply package.json]: seems like this was an auto add by something in vscode. removed it in 9804e9d

File: src/components/m-table-toolbar.js (hunk 4 above)
Comment (floAr) [reply src/components/m-table-toolbar.js]: done

File: src/components/m-table-toolbar.js (hunk 1 above)
Comment (theeranan) [reply src/components/m-table-toolbar.js:19]: thank you

**Comment by floAr** [comment]
@mbrn I am getting some errors with the prettier step:


```
remote: Permission to mbrn/material-table.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/mbrn/material-table/': The requested URL returned error: 403
```

Is this something on my side to fix?
